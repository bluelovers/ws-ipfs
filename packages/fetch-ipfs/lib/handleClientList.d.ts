import { ITSValueOrArray, ITSResolvable } from 'ts-type';
import { IIPFSClientAddresses } from '@bluelovers/ipfs-http-client';
import Bluebird from 'bluebird';
import { IPFS } from 'ipfs-core-types';
export type IUseIPFSInput = string | Partial<IPFS> | IIPFSClientAddresses;
export declare function handleClientList(useIPFS: ITSValueOrArray<IUseIPFSInput>, filter?: (ipfs: Partial<IPFS>) => ITSResolvable<boolean>): Bluebird<IIPFSPromiseApi[]>;
