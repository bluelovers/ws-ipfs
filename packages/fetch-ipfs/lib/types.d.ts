/// <reference lib="es2020" />
export interface PromiseFulfilledResult<T> {
    status: "fulfilled";
    value: T;
}
export interface PromiseRejectedResult {
    status: "rejected";
    reason: any;
}
export declare type PromiseSettledResult<T> = PromiseFulfilledResult<T> | PromiseRejectedResult;
