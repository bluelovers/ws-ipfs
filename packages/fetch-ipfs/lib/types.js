"use strict";
///<reference lib="es2020" />
Object.defineProperty(exports, "__esModule", { value: true });
//# sourceMappingURL=types.js.map