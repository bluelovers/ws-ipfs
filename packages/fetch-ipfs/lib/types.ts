///<reference lib="es2020" />

export interface PromiseFulfilledResult<T>
{
	status: "fulfilled";
	value: T;

	reason?: never;
}

export interface PromiseRejectedResult<E = any>
{
	status: "rejected";
	reason: E;

	value?: never;
}

export type PromiseSettledResult<T, E = any> = PromiseFulfilledResult<T> | PromiseRejectedResult<E>;

