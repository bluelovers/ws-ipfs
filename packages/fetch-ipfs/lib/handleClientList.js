"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.handleClientList = void 0;
const ipfs_http_client_1 = __importDefault(require("@bluelovers/ipfs-http-client"));
const bluebird_1 = __importDefault(require("bluebird"));
function handleClientList(useIPFS, filter) {
    return bluebird_1.default
        .resolve(useIPFS)
        .then(useIPFS => {
        if (!Array.isArray(useIPFS)) {
            return [useIPFS];
        }
        return useIPFS;
    })
        // @ts-ignore
        .map(async (ipfs) => {
        if (!ipfs) {
            return;
        }
        else if (typeof ipfs === 'string') {
            return (0, ipfs_http_client_1.default)(ipfs)
                .catch(e => null);
        }
        else if (filter && await filter(ipfs)) {
            return ipfs;
        }
        // @ts-ignore
        else if (typeof ipfs === 'object' && (ipfs.port || ipfs.host)) {
            return (0, ipfs_http_client_1.default)(ipfs)
                .catch(e => null);
        }
        else if (ipfs) {
            return ipfs;
        }
        return;
    })
        .filter((ipfs) => {
        if (ipfs) {
            if (filter) {
                return filter(ipfs);
            }
            return true;
        }
    });
}
exports.handleClientList = handleClientList;
//# sourceMappingURL=handleClientList.js.map