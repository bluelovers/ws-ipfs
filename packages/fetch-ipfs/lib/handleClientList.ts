import { ITSValueOrArray, ITSResolvable } from 'ts-type';
import ipfsClient, { IIPFSClientAddresses } from '@bluelovers/ipfs-http-client';
import Bluebird from 'bluebird';
import { IPFS } from 'ipfs-core-types';

export type IUseIPFSInput = string | Partial<IPFS> | IIPFSClientAddresses;

export function handleClientList(useIPFS: ITSValueOrArray<IUseIPFSInput>,
	filter?: (ipfs: Partial<IPFS>) => ITSResolvable<boolean>,
)
{
	return Bluebird
		.resolve(useIPFS)
		.then(useIPFS =>
		{
			if (!Array.isArray(useIPFS))
			{
				return [useIPFS];
			}

			return useIPFS
		})
		// @ts-ignore
		.map<IIPFSPromiseApi>(async (ipfs: IUseIPFSInput) =>
		{
			if (!ipfs)
			{
				return
			}
			else if (typeof ipfs === 'string')
			{
				return ipfsClient(ipfs)
					.catch(e => null)
			}
			else if (filter && await filter(ipfs as any))
			{
				return ipfs as IPFS
			}
			// @ts-ignore
			else if (typeof ipfs === 'object' && (ipfs.port || ipfs.host))
			{
				return ipfsClient(ipfs as IIPFSClientAddresses)
					.catch(e => null)
			}
			else if (ipfs)
			{
				return ipfs as IPFS
			}

			return
		})
		.filter<IPFS>((ipfs) =>
		{
			if (ipfs)
			{
				if (filter)
				{
					return filter(ipfs)
				}

				return true
			}
		})
		;
}
