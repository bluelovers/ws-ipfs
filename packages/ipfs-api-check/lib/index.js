"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var version_1 = require("./check/version");
Object.defineProperty(exports, "version", { enumerable: true, get: function () { return version_1.version; } });
var id_1 = require("./check/id");
Object.defineProperty(exports, "id", { enumerable: true, get: function () { return id_1.id; } });
var cat_1 = require("./check/cat");
Object.defineProperty(exports, "cat", { enumerable: true, get: function () { return cat_1.cat; } });
var get_1 = require("./check/get");
Object.defineProperty(exports, "get", { enumerable: true, get: function () { return get_1.get; } });
var add_1 = require("./check/add");
Object.defineProperty(exports, "add", { enumerable: true, get: function () { return add_1.add; } });
var ls_1 = require("./check/ls");
Object.defineProperty(exports, "ls", { enumerable: true, get: function () { return ls_1.ls; } });
var pin_1 = require("./check/pin");
Object.defineProperty(exports, "pin", { enumerable: true, get: function () { return pin_1.pin; } });
var object_1 = require("./check/object");
Object.defineProperty(exports, "object", { enumerable: true, get: function () { return object_1.object; } });
var refs_1 = require("./check/refs");
Object.defineProperty(exports, "refs", { enumerable: true, get: function () { return refs_1.refs; } });
var dag_1 = require("./check/dag");
Object.defineProperty(exports, "dag", { enumerable: true, get: function () { return dag_1.dag; } });
var config_1 = require("./check/config");
Object.defineProperty(exports, "config", { enumerable: true, get: function () { return config_1.config; } });
//# sourceMappingURL=index.js.map