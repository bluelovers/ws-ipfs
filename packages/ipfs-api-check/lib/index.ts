

export { version } from './check/version';
export { id } from './check/id';

export { cat } from './check/cat';
export { get } from './check/get';
export { add } from './check/add';
export { ls } from './check/ls';

export { pin } from './check/pin';
export { object } from './check/object';
export { refs } from './check/refs';
export { dag } from './check/dag';
export { config } from './check/config';
export { name } from './check/name';
export { key } from './check/key';
export { ping } from './check/ping';
