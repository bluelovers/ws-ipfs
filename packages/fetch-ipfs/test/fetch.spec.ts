import fetchIPFS from '../index';
import raceFetchIPFS from '../race';
import { filterList } from 'ipfs-server-list';
import publishToIPFSRace from '../lib/put/race';
import { readFileSync } from 'fs';
import { join } from 'path';
import { handleClientList } from '../lib/handleClientList';

describe(`fetch-ipfs`, () =>
{

	const testData = Object.freeze({
		cid: `Qmaisz6NMhDB51cCvNWa1GMS7LU1pAxdF4Ld6Ft9kZEP2a`,
		length: 32,
	})

	const apis = filterList('API');

	test(`fetchIPFS`, async (done) =>
	{

		let actual = await fetchIPFS(testData.cid);

		expect(actual).toHaveLength(testData.length);
		expect(actual).toMatchSnapshot();

		done();
	});

	test(`raceFetchIPFS`, async (done) =>
	{

		let actual = await raceFetchIPFS(testData.cid, apis);

		expect(actual).toHaveLength(testData.length);
		expect(actual).toMatchSnapshot();

		done();
	});

	test(`publishToIPFSRace`, async (done) =>
	{

		let data = await readFileSync(join(__dirname, 'res', 'demo.png'));
		let cid = `QmdPAhQRxrDKqkGPvQzBvjYe3kU8kiEEAd2J6ETEamKAD9`;

		let actual = await publishToIPFSRace(data, apis);

		expect(actual).toMatchSnapshot();

		expect(actual[0].status).toStrictEqual(`fulfilled`);

		let target = actual[0].value[0];

		expect(target).toMatchSnapshot();
		expect(target.cid.toString()).toStrictEqual(cid);
		expect(target.path).toStrictEqual(cid);

		done();
	});

	test(`handleClientList`, async (done) =>
	{

		let actual = await handleClientList(apis);

		expect(actual).toHaveLength(1);
		expect(typeof actual[0].cat).toStrictEqual('function');
		expect(typeof actual[0].get).toStrictEqual('function');

		expect(await actual[0].version()).toMatchSnapshot();

		done();
	});

})
