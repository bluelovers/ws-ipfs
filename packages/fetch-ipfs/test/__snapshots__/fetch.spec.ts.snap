// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetch-ipfs fetchIPFS 1`] = `
Object {
  "data": Array [
    72,
    101,
    108,
    108,
    111,
    32,
    102,
    114,
    111,
    109,
    32,
    73,
    80,
    70,
    83,
    32,
    71,
    97,
    116,
    101,
    119,
    97,
    121,
    32,
    67,
    104,
    101,
    99,
    107,
    101,
    114,
    10,
  ],
  "type": "Buffer",
}
`;

exports[`fetch-ipfs handleClientList 1`] = `
Object {
  "commit": "1f0de1aaf",
  "golang": "go1.12.13",
  "repo": "7",
  "system": "amd64/linux",
  "version": "0.4.23",
}
`;

exports[`fetch-ipfs publishToIPFSRace 1`] = `
Array [
  Object {
    "status": "fulfilled",
    "value": Array [
      Object {
        "cid": Object {
          "codec": "dag-pb",
          "hash": Object {
            "data": Array [
              18,
              32,
              223,
              132,
              95,
              120,
              239,
              37,
              73,
              105,
              194,
              216,
              47,
              210,
              17,
              220,
              60,
              70,
              37,
              120,
              237,
              137,
              129,
              233,
              243,
              67,
              167,
              8,
              53,
              32,
              216,
              26,
              53,
              116,
            ],
            "type": "Buffer",
          },
          "version": 0,
        },
        "path": "QmdPAhQRxrDKqkGPvQzBvjYe3kU8kiEEAd2J6ETEamKAD9",
        "size": 98748,
      },
    ],
  },
]
`;

exports[`fetch-ipfs publishToIPFSRace 2`] = `
Object {
  "cid": Object {
    "codec": "dag-pb",
    "hash": Object {
      "data": Array [
        18,
        32,
        223,
        132,
        95,
        120,
        239,
        37,
        73,
        105,
        194,
        216,
        47,
        210,
        17,
        220,
        60,
        70,
        37,
        120,
        237,
        137,
        129,
        233,
        243,
        67,
        167,
        8,
        53,
        32,
        216,
        26,
        53,
        116,
      ],
      "type": "Buffer",
    },
    "version": 0,
  },
  "path": "QmdPAhQRxrDKqkGPvQzBvjYe3kU8kiEEAd2J6ETEamKAD9",
  "size": 98748,
}
`;

exports[`fetch-ipfs raceFetchIPFS 1`] = `
Object {
  "data": Array [
    72,
    101,
    108,
    108,
    111,
    32,
    102,
    114,
    111,
    109,
    32,
    73,
    80,
    70,
    83,
    32,
    71,
    97,
    116,
    101,
    119,
    97,
    121,
    32,
    67,
    104,
    101,
    99,
    107,
    101,
    114,
    10,
  ],
  "type": "Buffer",
}
`;
